codes/c/chapter_hashing/array_hash_map.c

```diff
@@ -0,0 +1,215 @@
+/**
+ * File: array_hash_map.c
+ * Created Time: 2023-03-18
+ * Author: Guanngxu (446678850@qq.com)
+ */
+
+#include "../include/include.h"
+
+/* 哈希表默认数组大小 */
+# define HASH_MAP_DEFAULT_SIZE 100
+
+/* 键值对 int->string */
+struct Entry {
+    int key;
+    char *val;
+};
+
+typedef struct Entry Entry;
+
+/* 用于表示键值对、键、值的集合 */
+struct MapSet {
+    void *set;
+    int len;
+};
+
+typedef struct MapSet MapSet;
+
+/* 基于数组简易实现的哈希表 */
+struct ArrayHashMap {
+    Entry *buckets[HASH_MAP_DEFAULT_SIZE];
```

Comment:
Is it better to set MapSet as a member of ArrayHashMap? Then we don't have to pass *set to the iterator `void valueSet(ArrayHashMap *d, MapSet *set)`.

Reply:
`MapSet` just a tool `struct`. I think `MapSet` has nothing to do with `HashMap`, set `MapSet` as a member of `ArrayHashMap` isn't a good idea.